app/src/org/commcare/android/view/c3/AxisConfiguration.java

```diff
@@ -186,6 +182,11 @@ private JSONObject getAxis(String prefix) throws JSONException, InvalidStateExce
         JSONObject config = new JSONObject();
         boolean isX = prefix.equals("x");
 
+        // X and primary Y axis show by default, but not secondary y. Force them all to show.

```

Comment:
the fix is that this code now happens after the `if (!showAxes)` check above, rather than happening outside of this function and ignoring `showAxes`.